library/kotlin/src/io/envoyproxy/envoymobile/Counter.kt

```diff
@@ -0,0 +1,24 @@
+package io.envoyproxy.envoymobile
+
+import io.envoyproxy.envoymobile.engine.EnvoyEngine
+import java.lang.ref.WeakReference
+
+/**
+ * A stat.
+ *
+ * Current the supported stat type is counter, and it can increment.
+ */
+class Counter internal constructor(
+  private val envoyEngine: WeakReference<EnvoyEngine>,
+  private val elements: List<Element>
+) {
+
+  /**
+   * Increment the counter by the given count.
+   */
+  fun increment(count: Int) {
```

Comment:
Should we provide a default value of 1?

Reply:
usually we overload with `increment()` being `increment(1)`. In favor of that.

Reply:
Sounds good.